File: Leetcode/tree/965_UnivaluedBinaryTree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        if(root==null) return true;
        boolean left = (root.left==null)?true:root.val==root.left.val;
        boolean right = (root.right==null)?true:root.val==root.right.val;
        return left&&right&&isUnivalTree(root.left)&&isUnivalTree(root.right);
    }
}

File: Leetcode/tree/965_UnivaluedBinaryTree/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if not root: return True
        left = True if not root.left else root.val==root.left.val
        right = True if not root.right else root.val==root.right.val
        return left and right and self.isUnivalTree(root.left) and self.isUnivalTree(root.right)
